src/index.ts

```diff
@@ -137,32 +134,25 @@ export class EspLoader {
       throw ConnectError;
     }
 
-    await this.flushInput();
+    await this.reader.waitSilent(1, 200);
     await this.chipFamily();
   }
 
   private async try_connect(): Promise<boolean> {
-    await this.serialPort.setSignals({ dataTerminalReady: false });
-    await this.serialPort.setSignals({ requestToSend: true });
+    await this.serialPort.setSignals({ dataTerminalReady: false, requestToSend: true });
     await sleep(100);
-    await this.serialPort.setSignals({ dataTerminalReady: true });
-    await this.serialPort.setSignals({ requestToSend: false });
+    await this.serialPort.setSignals({ dataTerminalReady: true, requestToSend: false });
     await sleep(50);
-    await this.serialPort.setSignals({ dataTerminalReady: false });
+    await this.serialPort.setSignals({ dataTerminalReady: false, requestToSend: false });
 
     // Wait until device has stable output.
-    for (let i = 0; i < 20; i++) {
-      try {
-        await this.read(false, 1000);
-      } catch (e) {
-        if (e === TimeoutError) {
-          break;
-        }
-      }
-      await sleep(50);
+    const wasSilent = await this.reader.waitSilent(20, 1000);
+    if (!wasSilent) {
+      this.options.logger.debug("try_connect reader was not silent");
```

Comment:
I wonder if we should retry the bootloader startup step if this happens?

Reply:
Yes! Absolutely. Just a 5.repeat kind of thing

Reply:
IMO we should return here and say "Failed to enter bootloader". That could help us help the customer... Think GPy boards where they manually have to do something.

Reply:
I can write it out.